File: Data Structures & Algorithms/first-missing-positive/submission-3.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int size = nums.size() + 2;
        vector<int> arr(size, 0);

        for (int num : nums) {
            if (num >= 0 && num < size) {
                int index = std::min(num,size - 1);
                arr[index] = 1;
            }
        }

        for (int i = 1; i <= size; i++) {
            if (arr[i] == 0) {
                return i;
            }
        }
        
        return -1;
    }
};

File: Data Structures & Algorithms/verifying-an-alien-dictionary/submission-3.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        nums = []
        longest = 0

        for word in words:
            nums.append([])
            for char in word:
                num = order.index(char)
                nums[-1].append(num)
            longest = max(longest, len(word))

        print(nums)

        # go through them two by two, confirming they are in the correct order

        for i in range(len(nums) - 1):
            a = nums[i]
            b = nums[i+1]
            print(a)
            print(b)
            l = min(len(a), len(b))
            in_order = True
            non_equal = False
            for j in range(l):
                if a[j] > b[j]:
                    return False
                if a[j] < b[j]:
                    non_equal = True
                    break

            if non_equal is False and len(a) > len(b):
                return False

        return True
                


        